dvc/dependency/repo.py

```diff
@@ -112,23 +120,25 @@ def get_used_objs(
                 self.def_repo[self.PARAM_REV_LOCK] = rev
 
             path_info = PathInfo(repo.root_dir) / str(self.def_path)
-            try:
-                for odb, objs in repo.used_objs(
-                    [os.fspath(path_info)],
-                    force=True,
-                    jobs=kwargs.get("jobs"),
-                    recursive=True,
-                ).items():
-                    if odb is None:
-                        odb = repo.cloud.get_remote().odb
-                    self._check_circular_import(odb)
-                    used_objs[odb].update(objs)
-            except (NoRemoteError, NoOutputOrStageError):
-                pass
+            if not obj_only:
+                try:
+                    for odb, objs in repo.used_objs(
+                        [os.fspath(path_info)],
+                        force=True,
+                        jobs=kwargs.get("jobs"),
+                        recursive=True,
+                    ).items():
+                        if odb is None:
+                            odb = repo.cloud.get_remote().odb
+                            odb.read_only = True
+                        self._check_circular_import(objs)
+                        used_objs[odb].update(objs)
+                except (NoRemoteError, NoOutputOrStageError):
+                    pass
 
             try:
                 staged_obj = stage(
-                    local_odb,
+                    None,
```

Comment:
This will stage imported trees in the memfs ODB. We could also just stage them in the local ODB's staging area, but persisting them doesn't save us anything at the moment.

Since we re-clone the erepo across `dvc` command runs, we won't have a state entry for the import's (clone dir) path, so we will end up having to recompute the entire dir cache each time whether or not we already have it in the local ODB (or local staging ODB).

If/when we cache the import erepo clone dir, we could benefit from staging these trees locally, since if the cached clone dir hasnt changed, we would have a valid state entry for the imported path and could actually load the existing hash from local ODB (or local ODB's staging area)